crates/typst-syntax/src/parser.rs

```diff
@@ -1512,27 +1508,37 @@ struct Parser<'s> {
     /// Does include previously parsed trivia nodes, but does not include
     /// `current`.
     nodes: Vec<SyntaxNode>,
-    /// Stack of lexer modes to be pushed/popped. The current mode is implicitly
-    /// stored in the lexer.
-    modes: Vec<LexMode>,
-    /// Stack of newline modes to be pushed/popped. The current mode is the tail
-    /// of the vector.
-    newline_modes: Vec<NewlineMode>,
+    /// A mode for how we proceed on seeing a newline in Code mode.
+    nl_mode: AtNewline,
```

Comment:
I considered using `nl_stop` for a  bit, but came back to `nl_mode` recently. I'm not sure. I like the renaming of `NewlineMode` to `AtNewline`, but don't have a good name for both the variable holding that state and the `with_*` function that enters it. Thoughts?

Reply:
I don't have a good idea either. I think we can just leave it as-is. Though I think `AtNewline` works better for `Continue` and `Stop` than for the other three, since they are not verbs.

Reply:
I changed the names to `ContinueContextual`, `RequireColumn`, and `StopParBreak` on the last rebase. Thoughts?

Reply:
Sounds fine